presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1003,6 +1021,22 @@ public void dropTable(HiveIdentity identity, String databaseName, String tableNa
         }
     }
 
+    private static void deleteDir(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path, boolean recursive)
```

Comment:
We can call this `deleteDirRecursive` and remove the flag, since it's only called from one place.